core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/variable/listener/AbstractProblemPropertyInsensitiveVariableListener.java

```diff
@@ -0,0 +1,48 @@
+package org.optaplanner.core.impl.domain.variable.listener;
+
+import org.optaplanner.core.api.domain.variable.VariableListener;
+import org.optaplanner.core.api.score.director.ScoreDirector;
+import org.optaplanner.core.api.solver.change.ProblemChange;
+
+/**
+ * Used in situations where the variable listener does not read problem facts
+ * or any information in entities other than planning variables.
+ * This considerably improves performance of {@link ProblemChange}s.
+ *
+ * @param <Solution_>
+ * @param <Entity_>
+ */
+public abstract class AbstractProblemPropertyInsensitiveVariableListener<Solution_, Entity_>
```

Comment:
Rationale: many listeners would, when a problem property changed, immediately trigger `resetWorkingSolution(...)`.
This, in turn, triggers iteration over all entities, as is the case in many internal listeners.
These listeners now extend this class, and therefore all iteration is skipped, as the problem property change is irrelevant.

For backwards compatibility reasons, listeners not extending this (= actual users) get a default impl of these methods, and this will still call the working solution reset.

Reply:
As the class overrides any method related to problem facts, maybe the name should reflect that => `...ProblemFactInsensitive...`.